src/SetupApi/SetupApi.cs

```diff
@@ -219,6 +219,273 @@ public static partial class SetupApi
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff551104(v=vs.85).aspx"/>
+        [DllImport(nameof(SetupApi), SetLastError = true, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static unsafe extern bool SetupDiGetDeviceInstallParams(
+            SafeDeviceInfoSetHandle deviceInfoSet,
+            [Friendly(FriendlyFlags.In | FriendlyFlags.Optional)] SP_DEVINFO_DATA* deviceInfoData,
+            [Friendly(FriendlyFlags.Bidirectional)] SP_DEVINSTALL_PARAMS* deviceInstallParams);
+
+        /// <summary>
+        /// The <see cref="SetupDiSetDeviceInstallParams(SafeDeviceInfoSetHandle, SP_DEVINFO_DATA*, ref SP_DEVINSTALL_PARAMS)"/> function sets device installation parameters for a device information set
+        /// or a particular device information element.
+        /// </summary>
+        /// <param name="deviceInfoSet">
+        /// A handle to the device information set for which to set device installation parameters.
+        /// </param>
+        /// <param name="deviceInfoData">
+        /// A pointer to an <see cref="SP_DEVINFO_DATA"/> structure that specifies a device information element in <paramref name="deviceInfoSet"/>.
+        /// This parameter is optional and can be set to <see langword="null"/>. If this parameter is specified, <see cref="SetupDiSetDeviceInstallParams(SafeDeviceInfoSetHandle, SP_DEVINFO_DATA*, ref SP_DEVINSTALL_PARAMS)"/>
+        /// sets the installation parameters for the specified device. If this parameter is <see langword="null"/>, <see cref="SetupDiSetDeviceInstallParams(SafeDeviceInfoSetHandle, SP_DEVINFO_DATA*, ref SP_DEVINSTALL_PARAMS)"/>
+        /// sets the installation parameters that are associated with the global class driver list for <paramref name="deviceInfoSet"/>.
+        /// </param>
+        /// <param name="deviceInstallParams">
+        /// An <see cref="SP_DEVINFO_DATA"/> structure that contains the new values of the parameters. The <see cref="SP_DEVINSTALL_PARAMS.cbSize"/>
+        /// must be set to the size, in bytes, of the structure before this function is called.
+        /// </param>
+        /// <returns>
+        /// The function returns <see langword="true"/> if it is successful.
+        /// Otherwise, it returns <see langword="false"/> and the logged error can be retrieved with a call to <see cref="Marshal.GetLastWin32Error"/>.
+        /// </returns>
+        /// <seealso href="https://msdn.microsoft.com/en-us/library/windows/hardware/ff552141(v=vs.85).aspx"/>
+        [DllImport(nameof(SetupApi), SetLastError = true, CharSet = CharSet.Unicode)]
+        [return: MarshalAs(UnmanagedType.Bool)]
+        public static unsafe extern bool SetupDiSetDeviceInstallParams(
+            SafeDeviceInfoSetHandle deviceInfoSet,
+            [Friendly(FriendlyFlags.In | FriendlyFlags.Optional)] SP_DEVINFO_DATA* deviceInfoData,
+            ref SP_DEVINSTALL_PARAMS deviceInstallParams);
```

Comment:
Can we change this last parameter to `[Friendly(FriendlyFlags.Bidirectional)] SP_DEVINSTALL_PARAMS*` as well like the others?